src/librustc_mir/borrow_check/place_ext.rs

```diff
@@ -52,7 +50,12 @@ impl<'tcx> PlaceExt<'tcx> for Place<'tcx> {
                 }
             }
             Place::Base(PlaceBase::Static(static_)) => {
-                tcx.is_static(static_.def_id) == Some(hir::Mutability::MutMutable)
+                match static_.promoted {
```

Comment:
Eliminate this match by creating one big boolean expression by using `is_none()`